docs/software/ml/pytorch.md

```diff
@@ -0,0 +1,392 @@
+        | regex               | 2022.8.17 |
+        | requests            | 2.32.3 |
+        | safetensors         | 0.4.5 |
+        | sentencepiece       | 0.1.99 |
+        | sentry-sdk          | 2.22.0 |
+        | setproctitle        | 1.1.10 |
+        | setuptools          | 69.2.0 |
+        | six                 | 1.16.0 |
+        | smmap               | 5.0.0 |
+        | sympy               | 1.13.1 |
+        | tiktoken            | 0.4.0 |
+        | tokenizers          | 0.21.0 |
+        | torch               | 2.6.0 |
+        | torchaudio          | 2.6.0a0+d883142 |
+        | torchmetrics        | 1.5.2 |
+        | torchvision         | 0.21.0 |
+        | tqdm                | 4.66.3 |
+        | transformer_engine  | 2.3.0.dev0+dd4c17d |
+        | transformers        | 4.48.3 |
+        | triton              | 3.2.0+gitc802bb4f |
+        | typing_extensions   | 4.12.2 |
+        | urllib3             | 2.1.0 |
+        | versioneer          | 0.29 |
+        | wandb               | 0.19.9 |
+        | Werkzeug            | 3.0.4 |
+        | wheel               | 0.41.2 |
+        | wrapt               | 1.15.0 |
+        | zarr                | 3.0.1 |
+        | zipp                | 3.17.0 |
+
+
+[](){#ref-uenv-pytorch-how-to-use}
+## How to use
+
+There are two ways to access the software provided by the uenv, once it has been started.
+
+=== "the default view"
+
+    The simplest way to get started is to use the `default` file system view,
+    which automatically loads all of the packages when the uenv is started.
+
+    !!! example "test mpi compilers and python provided by pytorch/v2.6.0"
+        ```console
+        # start using the default view
+        $ uenv start pytorch/v2.6.0:v1 --view=default
+
+        # the python executable provided by the uenv is the default, and is a recent version
+        $ which python
+        /user-environment/env/default/bin/python
+        $ python --version
+        Python 3.13.0
+
+        # the mpi compiler wrappers are also available
+        $ which mpicc
+        /user-environment/env/default/bin/mpicc
+        $ mpicc --version
+        gcc (Spack GCC) 13.3.0
+        $ gcc --version # the compiler wrapper uses the gcc provided by the uenv
+        gcc (Spack GCC) 13.3.0
+
+        # exit the uenv
+        exit
+        ```
+
+=== "Spack"
+
+    The pytorch uenv can also be used as a base for building software with
+    Spack, because it provides compilers, MPI, Python and common packages like
+    hdf5.
+
+    [Check out the guide for using Spack with uenv][ref-building-uenv-spack].
+
+## Adding Python packages on top of the uenv
+
+Uenvs are read-only, and cannot be modified. However, it is possible to add Python packages on top of the uenv using virtual environments.
+
+!!! example "creating a virtual environment on top of the uenv"
+    ```console
+    # start the uenv
+    $ uenv start pytorch/v2.6.0:v1 --view=default
+
+    # create a virtual environment
```

Comment:
Same comment as above. Some of these comments are a but "obvious" bit might still be useful. Maybe worth making them an annotation instead?